### PR TITLE
storagetypes: use built-in `min` function

### PR DESCRIPTION
**Pull request by Juneezee**
### What this PR does

We can use the built-in `min` function since Go 1.21.

Reference: https://go.dev/ref/spec#Min_and_max

### Why we need it and why it was done in this way
The following tradeoffs were made:

The following alternatives were considered:

Links to places where the discussion took place: 

### Checklist

This checklist is not enforcing, but it's a reminder of items that could be relevant to every PR.
Approvers are expected to review this list.

- [ ] Design: A [design document](https://github.com/kubevirt/community/tree/main/design-proposals) was considered and is present (link) or not required
- [x] PR: The PR description is expressive enough and will help future contributors
- [x] Code: [Write code that humans can understand](https://en.wikiquote.org/wiki/Martin_Fowler#code-for-humans) and [Keep it simple](https://en.wikipedia.org/wiki/KISS_principle)
- [x] Refactor: You have [left the code cleaner than you found it (Boy Scout Rule)](https://learning.oreilly.com/library/view/97-things-every/9780596809515/ch08.html)
- [ ] Upgrade: Impact of this change on upgrade flows was considered and addressed if required
- [ ] Testing: New code requires [new unit tests](https://github.com/kubevirt/kubevirt/blob/main/docs/reviewer-guide.md#when-is-a-pr-good-enough). New features and bug fixes require at least on e2e test
- [ ] Documentation: A [user-guide update](https://github.com/kubevirt/user-guide/) was considered and is present (link) or not required. You want a user-guide update if it's a user facing feature / API change.
- [ ] Community: Announcement to [kubevirt-dev](https://groups.google.com/g/kubevirt-dev/) was considered

### Release note

```release-note
NONE
```

### CODE REVIEW

**Review by ShellyKa13 — COMMENTED** [review]
/lgtm

File: pkg/virt-launcher/virtwrap/converter/converter_test.go [hunk 1]
```diff
@@ -162,7 +161,7 @@ var _ = Describe("Converter", func() {
 			}
 			Convert_v1_Disk_To_api_Disk(context, &v1Disk, &apiDisk, devicePerBus, &numQueues, volumeStatusMap)
 			Expect(apiDisk.Capacity).ToNot(BeNil())
-			Expect(*apiDisk.Capacity).To(Equal(storagetypes.Min(capacity, requests)))
+			Expect(*apiDisk.Capacity).To(Equal(min(capacity, requests)))
```
[the remark this answers is not in the document]
Comment (Juneezee) [reply pkg/virt-launcher/virtwrap/converter/converter_test.go]: @akalenyu Thanks for reviewing the PR. I have updated this test to use an expected value in commit https://github.com/kubevirt/kubevirt/pull/13210/commits/abdb4cdc6e97855f3b4f35d1abc23bc9c9011e02

**Comment by Juneezee** [comment]
@ShellyKa13 @akalenyu friendly ping for approval :blush:

**Comment by Juneezee** [comment]
Rebased to resolve conflicts in `pkg/virt-launcher/virtwrap/converter/converter_test.go`.

**Review by ShellyKa13 — COMMENTED** [review]
/lgtm

**Pull request merged**